src/main/java/org/cardanofoundation/cfexploreraggregator/uniqueaccount/processor/UniqueAccountProcessor.java

```diff
@@ -37,27 +42,42 @@
 public class UniqueAccountProcessor {
 
     private final UniqueAccountRepository uniqueAccountRepository;
-    private final AddressInEpochRepository addressInEpochRepository;
+    private final UniqueAddressesInEpochRepository uniqueAddressesInEpochRepository;
 
     private final Set<String> uniqueAccounts = new ConcurrentSkipListSet<>();
 
 
     @EventListener
     @Transactional
     public void handleTransactionEvent(TransactionEvent event) {
-        event.getTransactions().forEach(tx -> tx.getUtxos().forEach(utxo -> {
-            try {
-            Address addr = new Address(utxo.getAddress());
-            if(addr.getDelegationCredential().isEmpty()) {
-                uniqueAccounts.add(utxo.getAddress());
+        event.getTransactions().forEach(tx -> tx.getUtxos().forEach(utxo -> processTransaction(utxo.getAddress())));
+    }
+
+    private void processTransaction(String address) {
+        if(address.startsWith("addr") || address.startsWith("stake")) {
+            Address addr = new Address(address);
+            if (addr.getAddressType() == AddressType.Base) {
+                String stakeAddress = AddressProvider.getStakeAddress(addr).getAddress();
+                uniqueAccounts.add(stakeAddress);
```

Comment:
I've seen this code before, this feels like it could be a smell, maybe we need a utility?

Could it be worth to add in CCL ? What are we trying to do here?